be/src/runtime/query_statistics.cpp

```diff
@@ -23,10 +23,28 @@
 
 namespace starrocks {
 
+void QueryStatistics::to_pb(PQueryStatistics* statistics) {
+    DCHECK(statistics != nullptr);
+    statistics->set_scan_rows(scan_rows);
+    statistics->set_scan_bytes(scan_bytes);
+    statistics->set_returned_rows(returned_rows);
+    statistics->set_cpu_cost_ns(cpu_ns);
+    statistics->set_mem_cost_bytes(mem_cost_bytes);
+    *statistics->mutable_stats_items() = {_stats_items.begin(), _stats_items.end()};
+}
+
 void QueryStatistics::merge(QueryStatisticsRecvr* recvr) {
     recvr->merge(this);
 }
 
+void QueryStatistics::merge_pb(const PQueryStatistics& statistics) {
+    scan_rows += statistics.scan_rows();
+    scan_bytes += statistics.scan_bytes();
+    cpu_ns += statistics.cpu_cost_ns();
```

Comment:
add or max ?

Reply:
It's used to merge statistic from other query fragments, so it should be `add` instead of max.

Reply:
if a plan is 
```
fragment 1
JOIN
|-EXCHANGE(id=1)
EXCHANGE 

fragment 2
SINK(dst=1)
|
JOIN
|-EXCHANGE(id=2)
|
EXCHANGE

fragment 3
SINK(dst=2)
|
AGG
```

real mem peak cost is max((fragment 1 + fragment 2), (fragment2, fragment3))

But I don't have a better idea

Reply:
In a complex plan, this algorithm has the potential to give a greater memory consumption statistic.

Reply:
Add TODO in description, and leave it for future work.